### ISSUE TITLE
Error in model building: dependencies from multivariate nodes

### ISSUE THREAD

**Issue by danielturek** [nimble-dev/nimble#340]
The following code produces an error, and I don't believe it should.

This error does **not** seem to be new, since merging refactor-modelDef.  I went back before that merge, and the error occurs just the same.

Note this error does not occur if the second declaration depends on x[1]:  `y[1] ~ dnorm(x[1], 1)`

```
library(nimble)
code <- nimbleCode({
    x[1:4] ~ dmnorm(mu[1:4], C[1:4,1:4])
    y[1] ~ dnorm(x[2], 1)
})
constants <- list(mu = rep(0,4), C = diag(4))
Rmodel <- nimbleModel(code, constants)

## defining model...
## Error in genExpandedNodeAndParentNames3(debug = debug) : 
##   Error building model, there are multiple definitions for nodes:x[1:4]
```

**Comment by perrydv** [comment]
This is a bug in the system that splits nodes into multiple vertices based on RHS expressions and determines the resulting vertex names.  Specifically `x[1:4]` should be split into `x[2]` and `x[1-s-4]` but determination of when the split is contiguous is incorrect and results in a vertex `x[1:4]` that duplicates the name of the original node `x[1:4]`.  That happens [here](https://github.com/nimble-dev/nimble/blob/e8cfb1240000fbfa1f4c8d70bdbf2c4c05f62ad1/packages/nimble/R/BUGS_modelDef.R#L1211)

More generally we would benefit from detailed unit tests of BUGS processing.  That has been a gap.

**Comment by danielturek** [comment]
Thanks for the quick explanation.  This is something that can be fixed, then?

**Comment by perrydv** [comment]
I think so, but it is a bit more involved than the other issue you just filed.  It would make sense to put in place detailed unit tests as part of fixing this bug.

**Comment by perrydv** [comment]
Fixed by PR #347
